### PR TITLE
Fixes:#5178

### PR DESCRIPTION
**Pull request by pramodgoud7**
Fixes: #5178

Fixed the bug by updating below code :

```
setupValidator(serverOptions) {
    if (!this.validatorCompiler) {
      this.validatorCompiler = this.getValidatorBuilder()(this.schemaBucket.getSchemas(), serverOptions.ajv);
    }
    if (this.validatorCompiler !== undefined && !this.addedSchemas) {
      return;
    }
    this.validatorCompiler = this.getValidatorBuilder()(this.schemaBucket.getSchemas(), serverOptions.ajv);
  }
```

image

### CODE REVIEW

File: my-app/.gitignore [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Logs
```
Comment (mcollina) [inline my-app/.gitignore:1]: I think all these files are not needed

**Review by mcollina — CHANGES_REQUESTED** [review]
There are a lot of not-needed files and tests are missing

File: my-app/.gitignore (hunk 1 above)
Comment (pramodgoud7) [reply my-app/.gitignore:1]: I understand that you think these files are not needed. Could you please provide some more context or information about why you think so? That would help me better understand your perspective and provide a more informed response

**Comment by Uzlopak** [comment]
#5188 by @derammo is trying to solve the same issue.
We did already some discussions in that PR.

Can you elaborate why your solution is preferrable over @derammo s PR?

File: my-app/.gitignore (hunk 1 above)
Comment (Eomm) [reply my-app/.gitignore:1]: `my-app/` is not needed by this project to run

**Comment by derammo** [comment]
I split out the test cases for #5178 to a separate branch, so you can merge them to test your PR

https://github.com/fastify/fastify/compare/main...derammo:fastify:derammo_5178_tests

**Comment by Fdawgs** [comment]
#5188 has fixed this issue. Thanks for the PR though!

**Pull request closed without merging**